shared/wallets/wallet/settings/container.js

```diff
@@ -83,5 +85,7 @@ export default compose(
       this.props.refresh()
     },
   }),
-  setDisplayName('Settings')
+  setDisplayName('Settings'),
+  safeSubmit(['onCurrencyChange'], ['currency']),
```

Comment:
Hmm the currency changing experience can be pretty bad on low bandwidth, so I do want some kind of safety – gonna try the second approach.

Reply:
Actually from playing around with it a bit, I think I prefer not using safe submit for this. I think if you change your display currency ten times, then you'd expect to see it change ten times. Safe submit seems to just introduce some harder to predict behavior. This could be revisited in the future though